File: 1068.py
# Introductory Problems : Weird Algorithm
#  https://cses.fi/problemset/task/1068/

# Basic : Just do as the question says.

n = int(input())
log = []
log.append(n)
while log[-1] != 1:
    if log[-1]%2 == 0:
        log.append(log[-1]//2)
    elif log[-1]%2 == 1:
        log.append((log[-1])*3+1)
res = [str(l) for l in log]
print(' '.join(res))

File: 1068.cpp
// Introductory Problems
// https://cses.fi/problemset/task/1068/

#include <iostream>
// #include <list>
using namespace std;

typedef long long lolo;

int main(){
    lolo n;
    cin >> n;
    while (n != 1) {
        cout << n << " ";
        if (n%2) n = n*3+1;
        else n/=2;
    }
    cout << "1\n";
}
